src/wallet/wallet.go

```diff
@@ -87,137 +125,416 @@ func NewWallet(wltName string, opts Options) (*Wallet, error) {
 	}
 
 	w := &Wallet{
-		Meta: map[string]string{
-			"filename": wltName,
-			"version":  version,
-			"label":    opts.Label,
-			"seed":     seed,
-			"lastSeed": seed,
-			"tm":       fmt.Sprintf("%v", time.Now().Unix()),
-			"type":     "deterministic",
-			"coin":     string(coin),
+		Meta: map[string]interface{}{
+			metaFilename: wltName,
+			metaVersion:  Version,
+			metaLabel:    opts.Label,
+			metaSeed:     seed,
+			metaLastSeed: seed,
+			metaTm:       fmt.Sprintf("%v", time.Now().Unix()),
+			metaType:     "deterministic",
+			metaCoin:     string(coin),
 		},
 	}
 
+	// Generate addresses
+	if _, err := w.GenerateAddresses(opts.AddressNum); err != nil {
```

Comment:
I removed this functionality already, it is in service, why is it back here

Reply:
Without this, if we want to create an encrypted wallet with addresses, we need to first create an encrypted wallet, then unlock the wallet, generate addresses, then lock wallet. And now we just need to set options.AddressNum, and the NewWallet method will generate addresses, then locking the wallet.

Reply:
Seems this is not necessary, I will remove this.